src/receive_imf.rs

```diff
@@ -662,12 +699,21 @@ pub(crate) async fn receive_imf_inner(
 ///   display names. We don't want the display name to change every time the user gets a new email from
 ///   a mailing list.
 ///
+/// * `is_partial_download`: the message is partially downloaded.
+///   We only know the email address and not the contact fingerprint,
+///   but try to assign the message to some PGP-contact.
+///   If we get it wrong, the message will be placed into the correct
+///   chat after downloading.
```

Comment:
What about unencrypted emails that are partially downloaded? They should be assigned to the email contact, not to some PGP contact with this email address. Maybe there is some code for this already and I just haven't seen it, otherwise we need a check for whether Content-Type is multipart/encrypted.

Reply:
But this is not about prefetch but about partial downloads? And for partial downloads, we fetch the full headers, which includes Content-Type:

```rust
const BODY_PARTIAL: &str = "(FLAGS RFC822.SIZE BODY.PEEK[HEADER])";
```

Reply:
Ah, wait, this is both about prefetch and about actual partial downloading.

What about this:
- Rename `is_partial_download` parameter to `lookup_pgp_contact_by_addr`
- In `receive_imf_inner()`, replace `is_partial_download.is_some()` with (pseudocode) `is_partial_download.is_some() && headers.get("Content-Type") == "multipart/encrypted"`

Reply:
@link2xt agrees if it's easy to do